### PR TITLE
64bit int inputs

### PR DESCRIPTION
**Pull request by ryandillinfelton**
Changes filterInt to use bigint to allow for signed 64bit inputs. Same max as `system_memory_bytes` in elasticsearch since it's a signed long.

### CODE REVIEW

File: src/generated/graphql.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export type Scalars = {
   Float: number;
   JSON: any;
   JSONObject: { [key: string]: any };
+  int64: any;
```
Comment (fijshion) [inline src/generated/graphql.ts:16]: should this be `int64: number;`?

File: src/generated/graphql.ts [hunk 2]
```diff
@@ -851,6 +855,10 @@ export type TagsResolvers<ContextType = any, ParentType extends ResolversParentT
   __isTypeOf?: IsTypeOfResolverFn<ParentType, ContextType>;
 };
 
+export interface Int64ScalarConfig extends GraphQLScalarTypeConfig<ResolversTypes['int64'], any> {
```
Comment (fijshion) [inline src/generated/graphql.ts:858]: is this used anywhere? I think `new GraphQLScalarType` might be needed to configure the scalar. Something like this https://www.apollographql.com/docs/apollo-server/schema/custom-scalars/#example-restricting-integers-to-odd-values

Comment (fijshion) [reply src/generated/graphql.ts:858]: maybe using this package would make it easier https://github.com/Urigo/graphql-scalars

**Comment by ryandillinfelton** [comment]
@fijshion Fixed up and ready for re-review

**Comment by fstavela** [comment]
@ryandillinfelton Does this need a change in HBI too? I'm still getting 500 on large numbers, for example: `/hosts?filter[system_profile][system_memory_bytes]=8161812809771164067`

**Comment by fijshion** [comment]
any reason you went with [apollo-type-bigint](https://github.com/Nberezhnoy/apollo-type-bigint) vs [graphql-scalars](https://github.com/Urigo/graphql-scalars)? `apollo-type-bigint` looks like a package from a single dev and is a little scary to depend on. `graphql-scalars` looks like it's more mature. I know the code is super simple but I'm generally hesitant to depend on really small projects like `apollo-type-bigint`.

**Comment by ryandillinfelton** [comment]
> @ryandillinfelton Does this need a change in HBI too? I'm still getting 500 on large numbers, for example: `/hosts?filter[system_profile][system_memory_bytes]=8161812809771164067`

IIRC when I was testing them together with this change it worked fine. Are you still getting that issue when running against xjoin-search with this change?

**Comment by ryandillinfelton** [comment]
> any reason you went with [apollo-type-bigint](https://github.com/Nberezhnoy/apollo-type-bigint) vs [graphql-scalars](https://github.com/Urigo/graphql-scalars)? `apollo-type-bigint` looks like a package from a single dev and is a little scary to depend on. `graphql-scalars` looks like it's more mature. I know the code is super simple but I'm generally hesitant to depend on really small projects like `apollo-type-bigint`.

I felt like it was a bit overkill to include a bit scalar library for just one simple scalar. I agree a project with more maintainers feels safer, but for a simple scalar like this that's just wrapping a built in JavaScript type I think it's quite unlikely that the code will need to change. Maybe if some Apollo change breaks it, but even that seems unlikely, and we could just migrate it then if that ever comes up.

**Comment by fijshion** [comment]
I don't think the size of the library is a concern. A few extra MBs is a drop in the bucket. You're right the concern over bug fixes or the code changing is minimal. My main concern with using single dev libraries is security. We tend to automatically upgrade packages, at least minor versions, which means a critical security vulnerability could be automatically added to our project through this package. Since apollo-type-bigint is not really used by many other people - it might go unnoticed for quite awhile.

**Comment by ryandillinfelton** [comment]
> I don't think the size of the library is a concern. A few extra MBs is a drop in the bucket. You're right the concern over bug fixes or the code changing is minimal. My main concern with using single dev libraries is security. We tend to automatically upgrade packages, at least minor versions, which means a critical security vulnerability could be automatically added to our project through this package. Since apollo-type-bigint is not really used by many other people - it might go unnoticed for quite awhile.

I see, that makes sense. I'll switch it over.

**Review by fijshion — APPROVED** [review]

**Pull request merged**